cognite/client/_api/data_modeling/instances.py

```diff
@@ -349,8 +366,13 @@ def _create_other_params(
         return other_params
 
     @classmethod
-    def _dump_instance_source(cls, source: ViewId) -> dict:
-        return {"source": source.dump(camel_case=True)}
+    def _dump_instance_source(cls, source: ViewIdentifier | View) -> dict:
+        instance_source: ViewIdentifier
+        if isinstance(source, View):
+            instance_source = source.as_id()
+        else:
+            instance_source = source
+        return {"source": ViewId.load(instance_source).dump(camel_case=True)}
```

Comment:
@erlendvollset I considered it to be better to leave the ViewId.load function as is rather than accept a View as parameter as it looks a bit ugly. But this part of the logic might have some more clever solutions.

Reply:
Agree this is probably better.